storm-core/src/jvm/org/apache/storm/tuple/Fields.java

```diff
@@ -50,8 +50,12 @@ public Fields(List<String> fields) {
     
     public List<Object> select(Fields selector, List<Object> tuple) {
         List<Object> ret = new ArrayList<>(selector.size());
-        for(String s: selector) {
-            ret.add(tuple.get(_index.get(s)));
```

Comment:
You can replace _index.get(s) with fieldIndex(s) which throws IllegalArgumentException.